### PR TITLE
cli: rename `jj file print` (formerly `jj cat`) to `jj file show`

### PR DESCRIPTION
**Pull request by ilyagr**
We now have two `cmd_show` in the repo. I think this one should become `cmd_file_show`, but this should be done uniformly over all the commands for consistency.

I did *not* keep `print` as an alias (I couldn't find a compelling reason to do it), but let me know if anyone feels like keeping it.

(This was discussed in https://github.com/martinvonz/jj/issues/3812. We should decide before the release next week whether or not to do this).

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Review by yuja — APPROVED** [review]

**Comment by ilyagr** [comment]
I'll wait until tomorrow to merge, in case anyone has comments or requests to wait longer.

**Comment by emilazy** [comment]
Thank you for this! I dislike the programmer terminology of “print” to mean “show” or “write”. I don’t think it’s very clear in general, but I especially wouldn’t expect it for something that reads from a file rather than outputting a string directly. I think this will be more approachable for users and intuitive in light of `jj show`.

**Comment by arxanas** [comment]
> I dislike the programmer terminology of “print” to mean “show” or “write”. I don’t think it’s very clear in general

agreed, no [`lpr(1)`](https://man.archlinux.org/man/lpr.1) integration → NOT a `print` command 👏

**Review by arxanas — APPROVED** [review]

**Pull request merged**
